src/engine/OrderBy.cpp

```diff
@@ -33,6 +33,19 @@ string OrderBy::asString(size_t indent) const {
   return os.str();
 }
 
+// _____________________________________________________________________________
+vector<size_t> OrderBy::resultSortedOn() const {
+  std::vector<size_t> sortedOn;
+  sortedOn.resize(_sortIndices.size());
+  for (const pair<size_t, bool>& p : _sortIndices) {
+    if (!p.second) {
+      // Only ascending columns count as sorted.
```

Comment:
That seems like a really good idea. Performance wise it might not make a huge difference though, as, as far as I remember right now, all internal sorting is always ascending, and only the order by solution modifier in the end does descending sorting.